src/Umbraco.Infrastructure/Runtime/FileSystemMainDomLock.cs

```diff
@@ -86,24 +97,29 @@ private void CreateLockReleaseFile()
             }
             catch (Exception ex)
             {
-                _log.LogError(ex, "Unexpected exception attempting to create lock release signal file {file}", _releaseSignalFilePath);
+                _logger.LogError(ex, "Unexpected exception attempting to create lock release signal file {file}", _releaseSignalFilePath);
             }
         }
 
         private void DeleteLockReleaseFile()
         {
-            while (File.Exists(_releaseSignalFilePath))
+            List<Exception> encounteredExceptions = new();
+            for (var i = 0; i < s_maxTriesRemovingLockReleaseSignalFile; i++)
             {
                 try
                 {
                     File.Delete(_releaseSignalFilePath);
+                    return;
                 }
                 catch (Exception ex)
                 {
-                    _log.LogError(ex, "Unexpected exception attempting to delete release signal file {file}", _releaseSignalFilePath);
+                    _logger.LogError(ex, "Unexpected exception attempting to delete release signal file {file}", _releaseSignalFilePath);
+                    encounteredExceptions.Add(ex);
                     Thread.Sleep(500);
```

Comment:
Wonder if we should do this
```suggestion
                    Thread.Sleep(500 * (i+1));
```

Reply:
94d84c05